qiskit_experiments/curve_analysis/curve_analysis.py

```diff
@@ -271,402 +72,40 @@ def _fit_params(cls) -> List[str]:
-        ``self._data`` method to filter data in the target series.
-
-        .. notes::
-            The target metadata properties to define each curve entry is described by
-            the class attribute __series__ (see `filter_kwargs`).
-
-        Args:
-            experiment_data: ExperimentData object to fit parameters.
-            data_processor: A callable or DataProcessor instance to format data into numpy array.
-                This should take a list of dictionaries and return two tuple of float values,
-                that represent a y value and an error of it.
-        Raises:
-            DataProcessorError: When `x_key` specified in the analysis option is not
-                defined in the circuit metadata.
-            AnalysisError: When formatted data has label other than fit_ready.
-        """
-        self.__processed_data_set = list()
-
-        def _is_target_series(datum, **filters):
-            try:
-                return all(datum["metadata"][key] == val for key, val in filters.items())
-            except KeyError:
-                return False
-
-        # Extract X, Y, Y_sigma data
-        data = experiment_data.data()
-
-        x_key = self.options.x_key
-        try:
-            xdata = np.asarray([datum["metadata"][x_key] for datum in data], dtype=float)
-        except KeyError as ex:
-            raise DataProcessorError(
-                f"X value key {x_key} is not defined in circuit metadata."
-            ) from ex
-
-        if isinstance(data_processor, DataProcessor):
-            ydata = data_processor(data)
-        else:
-            y_nominals, y_stderrs = zip(*map(data_processor, data))
-            ydata = unp.uarray(y_nominals, y_stderrs)
-
-        # Store metadata
-        metadata = np.asarray([datum["metadata"] for datum in data], dtype=object)
-
-        # Store shots
-        shots = np.asarray([datum.get("shots", np.nan) for datum in data])
-
-        # Find series (invalid data is labeled as -1)
-        data_index = np.full(xdata.size, -1, dtype=int)
-        for idx, series_def in enumerate(self.__series__):
-            data_matched = np.asarray(
-                [_is_target_series(datum, **series_def.filter_kwargs) for datum in data], dtype=bool
-            )
-            data_index[data_matched] = idx
-
-        # Store raw data
-        raw_data = CurveData(
-            label="raw_data",
-            x=xdata,
-            y=unp.nominal_values(ydata),
-            y_err=unp.std_devs(ydata),
-            shots=shots,
-            data_index=data_index,
-            metadata=metadata,
-        )
-        self.__processed_data_set.append(raw_data)
-
-        # Format raw data
-        formatted_data = self._format_data(raw_data)
-        if formatted_data.label != "fit_ready":
-            raise AnalysisError(f"Not expected data label {formatted_data.label} != fit_ready.")
-        self.__processed_data_set.append(formatted_data)
+    def _num_qubits(self) -> int:
```

Comment:
I don't think this should be in the base class, since it is a data-specific value stored in the analysis class state. Since it is only used in rb analysis, it should add this method just in that subclass to compute num qubits where needed.

Reply:
Make sense.